### PR TITLE
Changes Virology Rather Than Killing It

### PR DESCRIPTION
**Pull request by Higgin**
## About The Pull Request
God, alright, here we go. See HackMD here: https://hackmd.io/@Higgin/HJljdBuNp

Alternative proposal to #79849 addressing the big problems with virology. ~~If you need a HackMD for it, I'll put one together, but I made a comment on that PR and can make it pretty simple here.~~ its done

1. Makes viruses eventually self-cure as long as you're alive. If you can keep somebody from dying, they can develop immunity.
2. Makes it so you can sleep comfortably and be well-fed to slow and even potentially defeat viruses without a cure.
3. Makes it so more dangerous viruses can start self-curing faster. This means Space Ebola is going to burn itself out quicker if a person stays alive from the other effects.
4. Makes spaceacillin helpful in naturally curing viruses, period, but with declining effectiveness over 100 cycles.
5. Makes it so curing a virus naturally without being well-fed or having rode it out from the peak may allow you to be reinfected/not have natural immunity. 
6. Makes it so being well-fed is a much stronger protection against random virus spread. 
7. Makes it so bypasses_immunity stuff like fungal TB and heart failure isn't subject to any of this.
8. Makes it so using ~~antibiotics~~ spaceacillin jesus christ or being malnourished can make you lose your healing viruses too. Pay attention to what you put in your body.
9. ** Makes it so blood can ~~transmit resistances again, not just vaccines. It's been a hot minute, but it used to work like this.~~ blood now can cure a virus if the donor has a resistance, but it doesn't confer lasting immunity. You need to overcome the virus yourself, carry a constant supply of pure blood, or get a vaccine to get a lasting fix. 
10. ** makes severity a function of disease stats and all active symptoms - not just the highest severity of the active symptoms. 
11. ** makes it so you can nosell symptoms firing with spaceacillin or resting down to a minimum chance of cure_chance to avoid symptoms each cycle, declining over time, over 100 cycles for a given disease.
12. ** makes it so wearing protective equipment prevents you from spreading respiratory-spread diseases normally - not just on the cough/sneezing symptoms.
13. ** gives MDs virology access standard, paramedics and coroners virology access on skeleton crew. virologists also get pharmacy access.
14. ** makes bypasses_immunity advanced diseases always override non-bypasses_immunity advanced diseases and resist being overridden by other advanced diseases. Sentient Disease now has bypasses_immunity. Sentient Disease fans rejoice!
15. ** also gives SD a buffer of extra stealth points so it has a bit longer to build up instead of almost uniformly getting spotted and dying early.
16. ** viruses now scale their severity as a function of their max symptoms. There's a lot more room to get viruses of varying duration and severity by adding fewer symptoms now - so creating a tradeoff between stats (and good thresholds) and the duration of your virus.
17. ** a whole bunch of defines to control all of this stuff - most recently added a multiplier for symptom appearance frequency. 

MAJOR UPDATES: REBALANCING TOWARDS 50% LETHALITY
https://docs.google.com/spreadsheets/d/e/2PACX-1vQ8rqMYFsR1mYj_FGzVjTfcnAF7un-VofOByPxcCCQr6lOOF5fhUgZga0oA4Q5-7K4hr7fCV0jFdmd9/pubhtml#
[Viro Rework Rebalance Tests.pdf](https://github.com/tgstation/tgstation/files/13447208/Viro.Rework.Rebalance.Tests.pdf)

After a shitload of testing, makes some of the most reliable, transmissible killers into less-reliable threats. See the above graphs and pictures for demonstrations of exactly how this was tested and done. 

## Why It's Good For The Game

It sucks to be hard-stuck to needing chemistry and medical to deal with viruses that somebody can randomly blast out without a care in the world, then be left to sit around waiting to die or otherwise be unable to do anything as the max-level symptoms fire off on repeat. 

This should put curing and surviving viruses much more back in the hands of normal crew without always ending up at the chemistry front window, although that is still the fastest and most reliable way to get better.

This also nerfs healing viruses a bit, or makes them a bit less fire-and-forget if you fail to attend to your body. There's more I'd like to do in the future and potentially some of the other classic viruses that could use bypasses_immunity added, values tweaked, but for now - this seems like the best way to preserve virology as a level of depth and complexity in the game in a way that rewards people doing intuitive things to counterplay it when used harmfully.

This also puts more of the mid-range bad symptoms into a better place balance-wise because the worst ones pretty much only fire at max stages. With the way this works out, you bounce back and forth between the max stage and lower stages before, over time, trending towards a cure. Symptoms that provide more significant effects at lower stages now have a place that isn't totally overshadowed by the killdeath stage 5 ARDS + junk symptoms virus Dr. Ambatu Popov shat out in five minutes (as long as you survive the initial run-in with it.) 

## Changelog

:cl:
balance: most diseases can now be slowed, mitigated, and eventually cured through being well-fed, resting, and using spaceacillin. Curing diseases through this way will give you immunity if you experience them at their peak/maximum and aren't starving/malnourished when they cure.
balance: disease symptoms can be forestalled for up to 100 cycles with a declining chance of avoiding them over time using rest or spaceacillin.
balance: This does not apply to things like fungal TB; it does apply to healing viruses if you don't take care of yourself by staying fed and avoiding spaceacillin.
balance: disease can be cured through direct injection or ingestion of cured blood. However, curing disease in this way does not provide lasting immunity. You need to naturally beat the virus or get a vaccine for that.
balance: Wearing internals or using protective equipment while infected can limit the spread of respiratory illnesses from yourself to others. Contact transmission is still possible however.
balance: Medical Doctors now have roundstart virology access. Paramedics and coroners now get virology access on skeleton shift access. Virologists now have roundstart pharmacy access.
balance: Sentient Diseases now resist being overridden by other advanced diseases and can always override other advanced diseases; they also have an extra bonus on their stealth stat to help make up for early outing without a bit more testing.
balance: biohazard lockers now also contain a syringe of spaceacillin (in line with the orderable kit from cargo.)
balance: Virus severity is now also a function of the number of symptoms out of max your virus has. Experiment with different combinations using less than six symptoms to make viruses that are deceptively less-obvious and less quick to self-cure at the tradeoff of stats.
/:cl:

### CODE REVIEW

**Comment by EuSouAFazer** [comment]
> Makes it so using antibiotics or being malnourished can make you lose your healing viruses too. Pay attention to what you put in your body.

Antibiotics do nothing for viruses irl; you're thinking of antivirals, antibiotics is for bacterias. Improperly taking antibiotics for non-bacteria stuff contributes to the formation of super bacterias irl and it's not a cool thing to encourage in-game

**Comment by SinfulBliss** [comment]
> > Makes it so using antibiotics or being malnourished can make you lose your healing viruses too. Pay attention to what you put in your body.
> 
> Antibiotics do nothing for viruses irl; you're thinking of antivirals, antibiotics is for bacterias. Improperly taking antibiotics for non-bacteria stuff contributes to the formation of super bacterias irl and it's not a cool thing to encourage in-game

this may surprise you but generally it is not good practice to emulate things you see on spaceman

**Comment by DrDiasyl** [comment]
NEGATIVE 1 GBP, YOU'RE DOOMED MY FRIEND.

**Comment by Higgin** [comment]
> > Makes it so using antibiotics or being malnourished can make you lose your healing viruses too. Pay attention to what you put in your body.
> 
> Antibiotics do nothing for viruses irl; you're thinking of antivirals, antibiotics is for bacterias. Improperly taking antibiotics for non-bacteria stuff contributes to the formation of super bacterias irl and it's not a cool thing to encourage in-game

I'm not going to take us down into the weeds that far at this point. This is a space game, and most of the people who are lucky enough to be playing it are not in the places where MDRTB and MRSA are truly problems yet. Antibiotics in this instance fill the abstract role we're using them for as 'something the crew can access as a way to fight the bad thing' - and even IRL they do end up used in quite a few viral illnesses anyway as an approach to handling secondary or opportunistic infections like bacterial pneumonias.

I sympathize with what you're getting at. I do not consider it likely that I'm encouraging people to do a sub-extinction course of their meds by not adding in an extra hour or two to make sure it says 'DON'T THROW AWAY YOUR PILLS BEFORE YOU'RE DONE SO YOU DON'T GIVE MDR LATENT TB TO SOMEBODY WITH HIV PLEASE' on every spaceacillin syringe or saying 'oh this one's an antiviral.'

If I get back around to it or adding more varied, specific cures at some point, sure, but I'm not going to be doing this for now.

**Comment by HalcyonicWolf** [comment]
If it's any consolation to the pedants, spaceacilin in game is referred to as an Anti-Viral as well as on the wiki

**Comment by Krypandenej** [comment]
god's work. Makes a normally nigh-inescapable weapon mostly manageable by a normal crew easier to handle, but could conceivably still work to finish off a scattered, weakened and wounded crew. Maybe add some checks for lower health = slightly less resistance, with MUCH less resistance with bleeding and compound frac wounds?

**Comment by Higgin** [comment]
> god's work. Makes a normally nigh-inescapable weapon mostly manageable by a normal crew easier to handle, but could conceivably still work to finish off a scattered, weakened and wounded crew. Maybe add some checks for lower health = slightly less resistance, with MUCH less resistance with bleeding and compound frac wounds?

this'd be cool in the long run imo, i dig it

there's a ton that could be wrapped in, even potentially some paradoxical stuff for more 'radical' ways of getting rid of or changing viruses - the way Bay used to do it was you could nuke somebody with radium or other chems to potentially cure the virus (but usually killing their liver and fucking them up majorly in the process.)

Right now my biggest thought with the design doc this evening and putting this where it would be good to go, right now, is in making stats affect overall severity so a boosted 5 neutered + 1 active virus of level 1-2 symptoms isn't hanging around for ten minutes killing every lizard or hardstunning everyone with coughs totally unimpeded.

**Comment by Googles-Hands** [comment]
This is fine as long as the simple diseases (diseases that occur on their own and have custom effects, like Pierrots Throat), with the permanent cures are not affected by this.

**Comment by DaydreamIQ** [comment]
This kind of just seems like it'd completely fuck over the entire point of virology, if the viruses can cure themselves why bother taking the time to make one? Yeah you can just reinfect yourself (Assuming the person cured doesn't gain immunity which is also a really bad idea) but that just means people will start running around with pill bottles full of viruses instead of actually contributing to keeping it in their system

Also this would further put sentient disease in the dirt but lets be honest I don't think there's a coder alive who wants sentient disease to prosper anymore

File: code/datums/diseases/_disease.dm [hunk 1]
```diff
@@ -86,11 +93,100 @@
...
+	if(stage == max_stages && stage_peaked != TRUE) //mostly a sanity check in case we manually set a virus to max stages
+		stage_peaked = TRUE
+
+	if(stage_peaked && !(disease_flags & CHRONIC) && disease_flags & CURABLE && bypasses_immunity != TRUE)
+		if(stage == max_stages) //every cycle we spend at max stage counts towards eventually curing the virus
+			peaked_cycles += 1
+		switch(severity)
+			if(DISEASE_SEVERITY_POSITIVE) //good viruses don't go anywhere after hitting max stage - you can try to get rid of them by sleeping earlier
+				cycles_to_beat = DISEASE_CYCLES_POSITIVE
+				if(affected_mob.satiety > 0 || slowdown == 1) //any sort of malnourishment/immunosuppressant opens you to losing a good virus
```
Comment (CRITAWAKETS) [inline code/datums/diseases/_disease.dm:105]: An unintended consequence of satiety is that TRAIT_NOHUNGER makes you not have any.
Ethereals actually require that trait for their hunger system to work, so they wouldn't be able to sustain any positive virus!
I'm not sure that's intended...

**Comment by MrStonedOne** [comment]
> It sucks to be hard-stuck to needing chemistry and medical to deal with viruses that somebody can randomly blast out without a care in the world, then be left to sit around waiting to die or otherwise be unable to do anything as the max-level symptoms fire off on repeat.

https://github.com/tgstation/tgstation/pull/44798 it used to be one of a few chems and some of them were easy to get as crew rather than one of 40 randomly selected. 
image

**Comment by Jezzared** [comment]
Soup should give an extra bonus on top of well-fed.
I want the chef to make soup during outbreaks for that authentic sick in bed experience.

**Comment by deathrobotpunch** [comment]
well it’s better than no virology as they are trying to do now

**Comment by Loafin34** [comment]
new medical soup pr en route

**Comment by Namwty** [comment]
My only concern is that this was only brought up **after someone made PR about removing it, not before.**
To avoid mental gymnastics, it means that if this passes, virology will be forgotten and not tweaked for god knows how long until someone makes a new "remove virology" in next year or so. ( And we can accept it, some random crap will arise, requiring changes, as virology just hit's different )

With removal, it's basically gone and shit ( majority, a.k.a. no one needs to babysit mechanic that is basically hated by majority of players and probs coders,, with this... we still have it, but once it passes everyone will ignore it's existence, even if it still causes frustration. Gives "put a band aid" vibe this PR, even if it's objectively better than removal ( but in long term... maybe removal is simply better )

**Comment by DrAmazing343** [comment]
> My only concern is that this was only brought up **after someone made PR about removing it, not before.** To avoid mental gymnastics, it means that if this passes, virology will be forgotten and not tweaked for god knows how long until someone makes a new "remove virology" in next year or so. ( And we can accept it, some random crap will arise, requiring changes, as virology just hit's different )
> 
> With removal, it's basically gone and shit ( majority, a.k.a. no one needs to babysit mechanic that is basically hated by majority of players and probs coders,, with this... we still have it, but once it passes everyone will ignore it's existence, even if it still causes frustration. Gives "put a band aid" vibe this PR, even if it's objectively better than removal ( but in long term... maybe removal is simply better )

I think there's definitely a way to tweak this PR to ensure lethality of good viro tots with crazy viruses/randomly spawning death viruses while also making sure that players who are prepped with meds or spam suture/regen mesh will not just be granted a slow, agonizing death. This PR, in my mind, just ensures that rounds with no virologist at all aren't doomed to questionable chemist/meddocs failing to make the third virus cure in the row and getting numbers'd to death or having a 15-corpse pileup in the lobby.

Virologists still have content in SWIFTLY tackling viruses as well as being preemptive with antiviral beneficial viruses! Once those both are handled, it does sound from all I've heard of viro that they're lacking in a lot of content afterwards, but this PR on it's own doesn't seem to push the needle one way or another on actual virologist content.

**Comment by xzero314** [comment]
Awful solution. This is hardly a fix at all. Oh sorry did you spend an hour making that virus to help the crew? Well it will just get auto cured now. 

LEAVE VIROLOGY ALONE

**Comment by Higgin** [comment]
> Awful solution. Oh sorry did you spend an hour making that virus to help the crew?
> 
> Well it will just get auto cured
> 
> Why is virology under attack all of a sudden? This pr essentially deletes the job by virtue of making it pointless.
> 
> LEAVE VIROLOGY ALONE. LET ME MAKE MY METABOLISM BOOST VIRUS IN PEACE.

tell your boosted-ass patients to not starve themselves and it'll be fine

**Comment by SomeoneYouProbablyKnow** [comment]
Yea dawg don't call em antibiotics

**Comment by SomeoneYouProbablyKnow** [comment]
Ghetto methods of curing viruses are nifty. moo (or maybe someone else) had a good idea for an immune system value sort of allowing you to come in contanct with a few infected people before you youself get infected.
If you're set on making changes to viro instead of letting it walk off a cliff consider that idea as well.

**Comment by MTandi** [comment]
I love this much more than just removal. Nice integration with cooking.

I've also had an idea that you could fight certain symptoms with corresponding medicines.

Say coughing should actually damage your lungs, reducing natural healing, and you could heal the lungs with Polypyrylium Oligomers medicine from mushrooms, thus removing the symptom from your instance of the disease. Repeat for other symptoms (like heal headache with Mannitol) and make the disease die when you dealt with the symptoms one by one.

Please consider this suggestion.

**Comment by OrionTheFox** [comment]
> I love this much more than just removal. Nice integration with cooking.
> 
> I've also had an idea that you could fight certain symptoms with corresponding medicines.
> 
> Say coughing should actually damage your lungs, reducing natural healing, and you could heal the lungs with Polypyrylium Oligomers medicine from mushrooms, thus removing the symptom from your instance of the disease. Repeat for other symptoms (like heal headache with Mannitol) and make the disease die when you dealt with the symptoms one by one.
> 
> This would unlock more ghetto ways of curing and make the Botany involved. Please consider this suggestion.

Everyone knows you can just drink milk to cure all status effects

**Comment by xzero314** [comment]
> > Awful solution. Oh sorry did you spend an hour making that virus to help the crew?
> 
> > 
> 
> > Well it will just get auto cured
> 
> > 
> 
> > Why is virology under attack all of a sudden? This pr essentially deletes the job by virtue of making it pointless.
> 
> > 
> 
> > LEAVE VIROLOGY ALONE. LET ME MAKE MY METABOLISM BOOST VIRUS IN PEACE.
> 
> 
> 
> tell your boosted-ass patients to not starve themselves and it'll be fine

 Considering that symptom actually is balanced and has the side effect of making you hungry then it will be literally impossible to have it work out from now on. This isnt a good solution. This is just more "i died to a virus" cope. Please come up with something else. Dont kill a job that some people enjoy because you are coping

**Comment by Jezzared** [comment]
People keep saying "don't kill the job".

This is the alternative to _actually_ killing it fully. If yall want it to stay, then find a different solution and PR it yourselves. Because most of us seem to be in agreement that virology in its current form isn't engaging to be on the receiving end of. I think it can be made to work, but I also don't know shit for fuck about virology balance and can't code (or have the time to learn right now).

It's this or removal. Option three is you suggest something constructive instead of shouting to "leave it alone". Because consensus among the people who can change things is that it _isn't_ fine as it is.

**Comment by Momo8289** [comment]
My gripes with virology are mostly with the job itself, and less-so with the virus system. Sure, they can be annoying, but they present a solid non-antag threat to the station, and give medical something to deal with. In that regard, I think this pr is great, as it makes viruses more managable for the average crew member, especially on lowpop shifts without a virologist or any chemists. 

My problem with virology the job is that, in my opinion at least, it is almost entirely just an unecessary extra step in the process of dealing with a virus. The hardest part of dealing with a virus is making the cure and getting a blood sample. Obviously, how hard this is can vary depending on the cure, but regardless, this is a job that chemists are responsible for. The virologist doesn't even have pharmacy access (I think they might on meta, but so do parameds so its an exception. Or maybe I'm thinking of skeleton crew access?). The rest of the process is (supposed to be) handled by the virologist, who takes the blood sample from the chemists, brings it to viro, presses a button, and gives the vaccine to the chemists. Their job is to press one button. The role of button presser is given to the most isolated person on the station (by design), who most often has little to no awareness of what's happening on the rest of the station, and who's only other job (which I'll get to in a moment) lends itself heavily to getting into the zone and not noticing comms. Despite being the *virologist,* they are probably the least suited to dealing with a virus on the station. 

Their only other job responsibility is making a helpful virus, which is at best somewhat helpful (self-resp, the brain healing one, etc) and at worst potentially harmful (coma, annoying/harmful symptoms left un-neutered by a rookie). Regardless of how helpful the symptoms are, ultimately a good virus' main job is to prevent other harmful viruses from showing up/spreading, which makes it a job that is actively removing its own usefulness. Don't need the button pusher when there's nothing to make us need the button. 

With this pr, the button pusher is now even more pointless, since crew can better mitigate/prevent viruses on their own. Again, not a bad thing imo, I just think it further makes the virologist job useless. 

Don't want to talk all this shit about viro without giving any suggestions to fix it. There needs to be some kind of process that has to happen to make a vaccine beyond just pressing a button. Maybe some kind of genetics-esque minigame that requires some kind knowledge/skill (or honestly any interaction at all) beyond just pressing a button that exists only to (poorly) justify this one crewmember's paycheque. 

Not saying you need to put that in this pr, I just want to get my thoughts on it out there, and this seems like the best place to do it.

**Comment by SomeoneYouProbablyKnow** [comment]
> Don't want to talk all this shit about viro without giving any suggestions to fix it. There needs to be some kind of process that has to happen to make a vaccine beyond just pressing a button.

Is that what's needed? Sometimes less is more (and while i'm all for yeeting the whole virology system) i think removing just the job of viro can be a big improvement, not any mechanics but just the position of "virologist". Turning it from something a single player has to focus on to an outcropping of the general medbay gameplay loop. 

Just like some scientists do toxins but toxins is not a job on its own. It would be insane if we had a toxins job, it's too destructive and there's not enough content for people to regularly play it (and not just abuse it).

**Comment by TheVekter** [comment]
> > It sucks to be hard-stuck to needing chemistry and medical to deal with viruses that somebody can randomly blast out without a care in the world, then be left to sit around waiting to die or otherwise be unable to do anything as the max-level symptoms fire off on repeat.
> 
> #44798 it used to be one of a few chems and some of them were easy to get as crew rather than one of 40 randomly selected.
> 
> image

I keep seeing you bring this up but I don't know if it's because you want us to go back to letting people cure viruses with orange juice (which seems entirely too easy) or because you want us to limit cures to be more specific.

**Comment by SomeoneYouProbablyKnow** [comment]
It hardly matters if there are 10 or 40 of them. It doesn't change what you actually do against a virus. The chem machine can pump out infinite amounts of chemicals you just need someone to mix the right thing, someone with more chem hours will be faster because they don't have to open the wiki.

**Comment by optimumtact** [comment]
> My only concern is that this was only brought up **after someone made PR about removing it, not before.** 

happens all the time, i refer to it as fear driven development and it's a useful property that i quite often abuse.

**Comment by Higgin** [comment]
Going to move this back to a draft and add a to-do when I get home, can't seem to draft on mobile -

- clamp severity and make some function of resistance, stage speed, and transmission add to it for advanced diseases (since classic diseases just get handed theirs)
- make positive satiety have some degree of scaling additional or multiplicative bonus on recovery with other factors
- make it so bypasses_immunity prevents using direct blood-to-blood vaccination

**Comment by Higgin** [comment]
> > My only concern is that this was only brought up **after someone made PR about removing it, not before.**
> 
> happens all the time, i refer to it as fear driven development and it's a useful property that i quite often abuse.

A kick in the ass is a healthy thing. I've been sitting around wishcasting a lot of this for a while, but I was content to scoot along with it because mass death viruses are rare and a bit taboo on Manny, SD is a nothingburger even now (and if this goes forward, I'm either going to have to distinguish it with some flags and extra points or just lobby for its removal because it'll be subject to all the problems above if I don't - maybe it has a space because it eats threat?) and healing viruses are quietly beneficial to play with in a way that didn't bother me (even if they carry along the same poverty of interaction and smuggle in the existence of the harmful viruses.) 

It's good to put stuff into focus here.

**Comment by Momo8289** [comment]
> > Don't want to talk all this shit about viro without giving any suggestions to fix it. There needs to be some kind of process that has to happen to make a vaccine beyond just pressing a button.
> 
> Is that what's needed? Sometimes less is more (and while i'm all for yeeting the whole virology system) i think removing just the job of viro can be a big improvement, not any mechanics but just the position of "virologist". Turning it from something a single player has to focus on to an outcropping of the general medbay gameplay loop. 
> 
> Just like some scientists do toxins but toxins is not a job on its own. It would be insane if we had a toxins job, it's too destructive and there's not enough content for people to regularly play it (and not just abuse it).

Under the assumption that viro is being changed/"fixed", yes, that is what's needed imo. Now, *should* it be fixed? Honestly, I'm also better with just removing the job and having it be a toxins-style thing.

**Comment by Higgin** [comment]
yeah i'm ambivalent about virology being a role vs. being a system, but some people like to learn and sit around playing with it in the fortress of solitude, so i don't really mind it as much - making it a thing all MDs have access to would help remove some of the obstacles to treating viruses, so probably not a bad move in the long run, really

**Comment by Higgin** [comment]
Made a couple of fairly major changes -

**resting/spaceacillin now give some measure of, over-time, decreasingly guaranteed ability to nosell symptom activation.**

**spaceacillin effectiveness wears down over time (happy antimicrobial resistance week btw)**

**cured blood can be used as a temporary cure but not for lasting protection**

_none of this still applies to fungal tb_

**Comment by Higgin** [comment]
@EuSouAFazer i softened and came around on not making it say 'antibiotics' and my response to you was shitty, so i apologize for that - happy antimicrobial awareness week

I'm not adding antivirals or anything like that at this point, but AMR is a serious issue more people should be aware of, and it fucks people up everywhere

**Comment by MrStonedOne** [comment]
@Vect0r2 
> I keep seeing you bring this up but I don't know if it's because you want us to go back to letting people cure viruses with orange juice (which seems entirely too easy) or because you want us to limit cures to be more specific.

It solves multiple issues that were brought up.

One complaint was how somebody can make a fuck ton of viruses and than you have to figure out the cure for each.

Oh hey look 1 single pr created that problem, without it all 50 variants of the virus would be cured with 1 of 3 or 4 chems, some of them you can get without chemistry help. Hell you can just scattershot cures.

Don't want that as a virologist? You just have to forgo some of the more deadly symptoms to push it up to the harder to get cures.

There used to be *real* tradeoffs with your virus design. hard to cure or deadly, easy to spread or hard to detect.

It was also harder to create the virus you wanted, no isolating symptoms, no neutering symptoms. you sometimes had 3 or 4 ideas for what you were hoping for and then you went with what the ~~rng~~ wizard gods gave you.

Hell this was part of the whole point behind the 3 symptom random replacement system.

I like this pr's approach to virology, and with some extra work (re-add the after 3 symptom rng replacement, kill neutering symptoms thing) viro will be in a much better place.

I've seen new complaints come out of the wood work that its a boring job to play if you aren't antag, but they sound like excuses to remove it, not actual arguments, given how the argument applies to half the job on the stations if you just distill it down like that.

Jobs do not need to be as fun as non-antag as antag.

**Comment by TheVekter** [comment]
I agree with most of what you said here aside from being concerned that some random viruses will just be stupid easy to get rid of. I also _kind of_ want to see virologists be the go-to people to cure most of the more dangerous viruses instead of it mostly being chemistry's job, but from what you said, the harder to cure stuff will need virology's help, so that's not a big deal.

> I've seen new complaints come out of the wood work that its a boring job to play if you aren't antag, but they sound like excuses to remove it, not actual arguments, given how the argument applies to half the job on the stations if you just distill it down like that.
> 
> Jobs do not need to be as fun as non-antag as antag.

I agree, but they should at least be fun. If we remove positive viruses (which we should IMO, they're basically worse nanites) we should be ideally giving virologists something to do during normal gameplay, otherwise they're just going to be budget medical doctors that sometimes get to do cool station-saving stuff.

That being said, I don't hate the idea of having a specialist job where one medical doctor on the station is specialized in curing viruses. That might not be a bad route to take.

**Comment by Higgin** [comment]
@MrStonedOne 
> 
> There used to be _real_ tradeoffs with your virus design. hard to cure or deadly, easy to spread or hard to detect.
> 
> It was also harder to create the virus you wanted, no isolating symptoms, no neutering symptoms. you sometimes had 3 or 4 ideas for what you were hoping for and then you went with what the ~rng~ wizard gods gave you.

Virus Statistic Distribution at Level

unfortunately before I get into something like a full symptom rebalance (which this at least partially tries to address by making higher severity come from stacking multiple symptoms + stats) it's sort of a solved science

most of the best or 'good enough' stat boosting stats exist at level 1 and 2, and they're all bad. 3/5 of the level 7 symptoms + any 5 of those will kill people dead as fuck. same story with neutering them and adding the 2 current level 8 symptoms - you get plasma fixation (one of the best healing symptoms) + regenerative coma

if you get to the point of knowing what you're doing, the tradeoffs just really aren't there any more

**Comment by Higgin** [comment]
Because this occurred to me while I'm out -

Ideally, a virus that spreads through no contact, effort, or risk on the part of the virologist would leave you with lasting, direct consequences no more than 50% of the time assuming you do absolutely nothing about it. 

Anything that raises that rate should only be at the expenditure of extra contact, effort, and risk by the virologist.

The relationship, then, between a disease's transmissibility and likelihood of lasting effect, however that appears, should be inverse. 

Since any disease might be spread inadvertently, though, and as long as highly lasting/serious symptoms can be transmissible, the rate of recovery for the most severe, quick-acting, serious disease should still probably be a natural coinflip in order for virology to not suck.

Making it so a boosted ARDS virus will kill a perfectly immobile, unresponsive person about 50% of the time, or cure itself, is going to be my target tonight, and though there's a lot more that could be done methodologically, I'll set the numbers according to that target for now. The scope of this is already getting wide, but it should end up after this somewhere on its face 'balanced.'

**Comment by Higgin** [comment]
image
This is my test virus. It's probably the fastest irrecoverably deadly thing you can make under the current system. Everything you see tested will be with this on 16 monkeys in a similar format to this - 50u/50u/50u/10u blood smoke grenade, in a contained environment where they can't fuck with each other. I'll post my tests and the defined values with some options tonight.

image

**Comment by CRITAWAKETS** [comment]
>test virus

you can neuter fever on this one for even more damage since otherwise fever and chills sorta cancel eachother out, and chills is generally more effective

**Comment by Higgin** [comment]
> > test virus
> 
> you can neuter fever on this one for even more damage since otherwise fever and chills sorta cancel eachother out, and chills is generally more effective

I was right in the middle of writing "some fucking pedant is going to say 'but itching does brute' and 'but chills and fever counteract each other' and this was uncannily timed.

**Comment by Higgin** [comment]
image
Test on current commit: 
1/16 survived. 
3/16 developed natural immunity.
2/3 died from being too far into crit when they finally got immune.

**Comment by deathrobotpunch** [comment]
even though they all died the fact that some of them had immunity means that people could harvest their blood and create vaccines.

**Comment by Higgin** [comment]
> even though they all died the fact that some of them had immunity means that people could harvest their blood and create vaccines.

or even use the blood directly for a cure, though, doing it in a crowded setting like that will result in just getting immediately reinfected

**Comment by Higgin** [comment]
Couple of highlights from testing - **see the links in the OP.**

_I trusted you..._

image

_Survive this shit and still get appendicitis._
image

_If you survive long enough from a death virus to catch a floor virus from the rotting buddies of your friends, maybe that's a good thing? (those viruses are also self-curing in this picture btw)_
image


_We made it wrong, as a joke!_ or, the ARDS virus that, when it doesn't hit its two thresholds, goes from nearly 100% lethal to 100% nonlethal (in an earlier build.) 
image

**Comment by Higgin** [comment]
Just as a note to anyone looking at TM results - while it should work fine with #79873, it's balanced for those symptoms at the top-end. I would recommend comparing tests with and without those symptoms in play.

**Comment by Higgin** [comment]
Current TM commit is acbcb7d. Healing viruses are being fucky and won't advance or recede if you aren't hungry (side-effect of cleaning up the code earlier: it was returning earlier than it should without trying to progress.) I didn't notice this during testing with my subjects being hungry. 585b77a fixes it.

**Comment by Higgin** [comment]
Couple of thoughts on recent changes:

- health and wellness are medical's whole deal - anything integral to health should be at some level basically accessible to medical line staff, so viro having their own access-locked shutaway doesn't make sense until there's something separate from that business for them to do. giving everyone access to viro at some point also reduces some of the current problem of 'see bad virus, lynch viro' too.
- SD has been criticized as a nothingburger threat sink and rightly so. making it not die immediately to contagious healing viruses, bypass them to infect, and ignore these changes to natural recovery - as a special thing - both makes SD more distinctive and less of a wet fart.

**Comment by Higgin** [comment]
Added a define to change symptom frequency (it just multiplies the random delay until a symptom can fire again, I left it at 1, but it's there now.) 

Slightly reduced the floor for cure_chance to 1 (with the multiplier, it's 3% per cycle to cure after having hit stage 1 with a virus that has 18 resistance - the way this works in practice is that you'll hang around stages 1-2 for longer, still contagious, but not getting the worst effects of a very high-resistance virus before it cures.)

Also made severity a function of the proportion of symptoms you have vs. total possible symptoms. It opens up a lot of room for non super-boosted-kill-death viruses to experiment and play around with in the middle.

**Pull request merged**